src/platform/plugins/shared/share/server/unused_urls_task/register_delete_unused_urls_route.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { Duration } from 'moment';
+import { IRouter, Logger, ReservedPrivilegesSet } from '@kbn/core/server';
+import { CoreSetup } from '@kbn/core/server';
+import { runDeleteUnusedUrlsTask } from './task';
+
+export const registerDeleteUnusedUrlsRoute = ({
+  router,
+  core,
+  urlExpirationDuration,
+  urlLimit,
+  logger,
+}: {
+  router: IRouter;
+  core: CoreSetup;
+  urlExpirationDuration: Duration;
+  urlLimit: number;
+  logger: Logger;
+}) => {
+  router.post(
+    {
+      path: '/internal/unused_urls_task/run',
+      security: {
+        authz: {
+          requiredPrivileges: [ReservedPrivilegesSet.superuser],
+        },
+      },
+      options: {
+        access: 'internal',
+        summary: 'Runs the unused URLs cleanup task',
+      },
+      validate: {},
+    },
+    async (_ctx, _req, res) => {
+      const { deletedCount } = await runDeleteUnusedUrlsTask({
```

Comment:
nit: On some of our larger/older clusters I expect most network requests to time out on this route... Maaaybe we should just make it start the task and then observe progress by polling?

Reply:
Note that this route is internal and mainly added for development and testing. I thought the timeout issue is acceptable and not worth complicating this further.